### DISCUSSION TITLE
Invoking Animation Node Functions from Another Script

### DISCUSSION THREAD

**Discussion by mikeashelby** [JacquesLucke/animation_nodes#1713] [Q&A] [answered]
Hi,

I'm just starting out using animation nodes, and it's an incredible addition to blender. Thank you for all your work on it!

I have a workflow whereby I import and update a series of text fields from an external source using a command-line script. I'd love to use animation nodes to be able to animate these in a fancy way (per character animation being one example) using animation nodes.

Would it be possible to update the text objects, and then invoke — from my script — the update function on all the separate text object nodes?

I've taken a look at the source and my understanding is that I'd need to run the `updateSeparation` function from the `SeparateTextObjectNode` class. I'm just not quite sure how I'll get the relevant instance.

Any tips would be much appreciated!

Many thanks,

Mike

p.s. I'm currently running all this in blender 2.79 — not sure if that makes a difference!

**Comment by OmarEmaraDev** [comment, marked as answer]
If you can use a new version of Blender and Animation Nodes, the Decompose Text node might be a better option as it doesn't require explicit updates. See https://docs.animation-nodes.com/tutorials/decompose_text/.

In any case, you can do `bpy.data.node_groups[<treeName>].nodes[<nodeName>].updateSeparation()`.

**Reply by mikeashelby** [reply]
Amazing! That's very handy! Thank you so much for the speedy reply — I'll give that a whirl!
